### PR TITLE
feat: Bid voting modal

### LINKED ISSUE #1116: Related Bids & Tenders pop up
- The modal will be triggered after the user selects and voting option and clicks on Vote
- The vote won't be casted until the user sees this modal and clicks on the confirmation buttons (Similar to the Sentiment module)
- The modal will have to states, a pre cast and a post cast

Image

Image

Note that on the success instance only the proposals the user hasn't voted on should be displayed

Image

If the user effectively voted on all proposals, the success modal should look a bit different

### PR DESCRIPTION
**Pull request by ncomerci**
This closes #1116

### CODE REVIEW

**Review by 1emu — APPROVED** [review]

File: src/components/Modal/BidVotingModal/BidVotingModal.css [hunk 1]
```diff
@@ -0,0 +1,11 @@
+.BidVotingModal .BidVotingModal__Description > p {
```
Comment (andyesp) [inline src/components/Modal/BidVotingModal/BidVotingModal.css:1]: You can use the componentsClassNames prop in the Markdown component to modify all the paragraphs there.

File: src/components/Proposal/View/ProposalCard.tsx [hunk 2]
```diff
@@ -17,21 +19,36 @@ interface Props {
...
+  onCardClick,
+  isDisabled,
+}: Props) {
   const t = useFormatMessage()
-  const { id, title, user, start_at, finish_at } = proposal
+  const { id, title, user, start_at, finish_at, type, configuration } = proposal
   const { votes } = useProposalVotes(id)
   const dateText = useProposalDateText(start_at, finish_at)
+  const { formatNumber } = useIntl()
+  const hasBudget = type === ProposalType.Grant || type === ProposalType.Bid
```
[the remark this answers is not in the document]
Comment (andyesp) [reply src/components/Proposal/View/ProposalCard.tsx:40]: If we reuse this logic somewhere else I'd move it...

[the remark this answers is not in the document]
Comment (andyesp) [reply src/components/Proposal/View/ProposalCard.tsx:40]: Maybe this conditions can be in the `showBudget` variable?

**Review by andyesp — APPROVED** [review]

File: src/components/Proposal/View/ProposalCard.tsx [hunk 3]
```diff
@@ -17,34 +20,83 @@ interface Props {
...
     <Link
       className={classNames(
         'ProposalCard',
-        highlight && (isOverBudget ? 'ProposalCard--overbudget' : 'ProposalCard--highlight')
+        highlight && (isOverBudget ? 'ProposalCard--overbudget' : 'ProposalCard--highlight'),
+        isDisabled && 'ProposalCard--disabled'
       )}
-      href={locations.proposal(id)}
+      href={proposalUrl(id)}
+      target="_blank"
```
Comment (1emu) [inline src/components/Proposal/View/ProposalCard.tsx:56]: I think our link component already handles this

**Review by 1emu — APPROVED** [review]

File: src/components/Proposal/View/ProposalCard.tsx (hunk 3 above)
Comment (andyesp) [reply src/components/Proposal/View/ProposalCard.tsx:56]: It's a "local link", we need to specify it.

**Review by andyesp — APPROVED** [review]

**Pull request merged**